gammapy/spectrum/results.py

```diff
@@ -503,31 +686,83 @@ def plot_spectrum(self, energy_unit='TeV', flux_unit='cm-2 s-1 TeV-1',
         ax0.legend(numpoints=1)
         return gs
 
+    def plot_residuals(self, ax=None, energy_unit='TeV', **kwargs):
+        """Plot residuals
+
+        Parameters
+        ----------
+        ax : `~matplolib.axes`, optional
+            Axis
+        energy_unit : str, `~astropy.units.Unit`, optional
+            Unit of the energy axis
+
+        Returns
+        -------
+        ax : `~matplolib.axes`, optional
+            Axis
+        """
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        kwargs.setdefault('fmt', 'o')
+
+        y, y_err = self.residuals()
+        x = self.flux_points['energy'].quantity
+        x = x.to(energy_unit).value
+        ax.errorbar(x, y, yerr=y_err, **kwargs)
+
+        xx = ax.get_xlim()
+        yy = [0, 0]
+        ax.plot(xx, yy, color='black')
 
-class SpectrumFitResultDict(dict):
-    """Dict of several `~gammapy.spectrum.SpectrumFitResults`"""
+        ax.set_xlabel('E [{}]'.format(energy_unit))
+        ax.set_ylabel('Residuals')
+
+        return ax
+
+
+class SpectrumResultDict(OrderedDict):
+    """Dict of several spectrum results
+
+    * `~gammapy.spectrum.results.SpectrumStats`
+    * `~gammapy.spectrum.results.SpectrumFitResult`
+    """
 
     def info(self):
         raise NotImplementedError
 
-    def to_table(self):
-        """Create overview `~astropy.table.QTable`"""
+    @classmethod
+    def from_files(cls, files, identifiers=None):
+        """Create `~gammapy.spectrum.SpectrumResultDict` from a list of files
+
+        Parameters
+        ----------
+        files : list, tuple
+            Files to load
+        identifiers : list
+            Analysis identifiers
+        """
+        val = cls()
+        identifiers = files if identifiers is None else identifiers
+        for f, id in zip(files, identifiers):
+            temp = SpectrumResult.from_all(f)
+            val[id] = temp
+        return val
+
+    def to_table(self, **kwargs):
+        """Create overview `~astropy.table.Table`"""
 
         val = self.keys()
-        analyses = Column(val, name='Analysis')
-        table = self[val.pop(0)].to_table()
-        # Add other analysis with correct units
+        analyses = Column(val, name='analysis')
+        l = list()
         for key in val:
-            temp = QTable(self[key].to_table())
-            new_row = list()
-            for col in table.columns:
-                new_row.append(temp[col].to(table[col].unit).value)
-            table.add_row(new_row)
-
+            l.append(self[key].to_table(**kwargs))
+        table = vstack(l, join_type='outer')
         table.add_column(analyses, index=0)
```

Comment:
With Python 3 and Astropy and Gammapy master I'm getting this error when I follow along the new tutorial:
```
$ gammapy-spectrum display crab_analysis/spectrum_stats.yaml spectrum_stats_23592.yaml
Traceback (most recent call last):
  File "/Users/deil/Library/Python/3.4/bin/gammapy-spectrum", line 9, in <module>
    load_entry_point('gammapy', 'console_scripts', 'gammapy-spectrum')()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/deil/code/gammapy/gammapy/scripts/spectrum.py", line 99, in display_spectrum
    t = master.to_table(format=format)
  File "/Users/deil/code/gammapy/gammapy/spectrum/results.py", line 762, in to_table
    table.add_column(analyses, index=0)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/table/table.py", line 1387, in add_column
    self.add_columns([col], [index], rename_duplicate=rename_duplicate)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/table/table.py", line 1495, in add_columns
    self._init_from_cols(newcols)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/table/table.py", line 685, in _init_from_cols
    lengths = set(len(col) for col in cols)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/table/table.py", line 685, in <genexpr>
    lengths = set(len(col) for col in cols)
TypeError: len() of unsized object
```